metabase/metabase
# Add action routes

Epic #27867 (more broadly #27581)

Adds dedicated routes to the action editor instead of opening them inline inside the model detail page:

* new actions can be created from `/model/:modelId/detail/actions/new`
* existing actions can be edited from `/model/:modelId/detail/actions/:actionId`

### To Verify

1. Enable actions for your Postgres / MySQL database on the `/admin/databases/:id` page
2. Open `/model/:modelId/detail/actions`
3. Click "New action"
4. Ensure the editor did open in a clean state, and the URL is `/model/:modelId/detail/actions/new`
5. Ensure the model's database is preselected by default
7. Ensure you can create the action (name, description, query, model, form settings)
8. Click the pencil icon next to your question
9. Ensure the editor did open and shows action info, and the URL is `/model/:modelId/detail/actions/:actionId`
10. Ensure you can update the action
11. Ensure both "new" and "edit" links open if you just put them in an address bar


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/28063)
